services/redshift/drivers/RedshiftCluster.py

```diff
@@ -15,7 +15,7 @@ def __init__(self, cluster, rsClient):
         # print(self.cluster)
         return
     
-    def _checkPubliclyAcessible(self):
+    def _checkCluster(self):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this function is 16. By comparison, 98% of the functions in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the function has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code blocks such as on lines 25-31, lines 34-40, lines 43-49, lines 52-58, lines 61-67 into separate functions.